week2/homework/src/index.js

```diff
@@ -1,3 +1,97 @@
 'use strict';
 
-// TODO: Write the homework code in this file
+const fs = require('fs');
+const fileName = `${__dirname}/todoList.txt`;
+
+const command = process.argv[2];
+const args = process.argv.slice(3);
+
+const newItem = process.argv.slice(4);
+
+const number = parseInt(args);
+
+if (command === 'add') {
+  add(...args).then(() => console.log(`a new item has been added`));
+} else if (command === 'help' || command === undefined) {
+  help();
+} else if (command === 'list') {
+  list().then(data => console.log(`To-Dos:\n${data}`));
+} else if (command === 'remove') {
+  remove(number);
+} else if (command === 'reset') {
+  reset().then(() => console.log('All to-do items has been deleted'));
+} else if (command === 'update') {
+  update(number, newItem);
+}
+
+async function update(index, text) {
+  try {
+    const data = await list();
+    let arr = data.split('\n');
+    write(arr);
+    if (index <= arr.length && index > 0) {
+      arr.splice(index - 1, 1, text);
+      write(arr).then(console.log('an item has been removed'));
+    } else {
+      console.log('invalid number');
+    }
+  } catch (err) {
+    console.error(err);
+  }
+}
+
+async function remove(index) {
+  try {
+    const data = await list();
+    let arr = data.split('\n');
+    if (index <= arr.length && index > 0) {
+      arr.splice(index - 1, 1);
+      write(arr).then(console.log('an item has been modified'));
+    } else {
+      console.log('invalid number');
+    }
+  } catch (err) {
+    console.error(err);
+  }
+}
+
+function write(arr) {
+  return new Promise((resolve, reject) =>
+    fs.writeFile(fileName, `${arr.join('\n')}`, (err, data) => (err ? reject(err) : resolve(data))),
```

Comment:
No need for template-strings here, it can be written simply `arr.join('\n')`

Reply:
Hi Samir, Very big thanks for you. Each time you correct my homework i learn new something in coding.
I've modified all issues you mentioned here, you can have a look if you like. 
Thanks again.